### PR TITLE
Add answers to the assignment questions

### PR DESCRIPTION
**Pull request by leharbhatt**
I have added the answers to the questions in answers.txt. Please review and let me know what you think of it.

### CODE REVIEW

File: answers.txt [hunk 1]
```diff
@@ -1,4 +1,16 @@
 What is a gitignore file and what should it include for a Java project?
+
+The role of a .gitignore file is to tell git which files it should ignore before we make a commit. For JAVA files it should include compiled class file(*.class), log file (*.log), BlueJ files(*.ctxt), Mobile tool for Java files(.mtj.tmp/), Package files(*.jar, *.war, *.nar, *.ear ,*.zip , *.tar.gz, *.rar) and virtual machine crash logs(hs_err_pid*).
+
 What does fast-forward in Git mean?
+
```
Comment (ishansarangi) [inline answers.txt:6]: One thing to add is that, fast-forward merge happens only if master doesnt have any separate commits  i.e, it is not diverged.

File: answers.txt [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+Fast-forward in git will update only the branch pointer without creating a merge commit. In other words, instead of creating a new commit, git will just point master to the latest commit of the feature branch. 
+
 Why should you have different branches?
+
+Because having different branches is awesome. A Git branch is essentially an independent line of development. You can take advantage of branching when working on new features or bug fixes because it isolates your work from that of other team members. Since the branches are independent you don't have to worry about the changes that other developers are making.
+
+
 In a project how many remote Git repositories do you usually have?
```
Comment (ishansarangi) [inline answers.txt:14]: In one project we can have multiple repo. Suppose if we have two repos which are related i.e, one uses complete or some part of the other, then we can keep it in one.

Comment (leharbhatt) [reply answers.txt:14]: Corrected the answer, committed the changes and created a new pull request.

File: answers.txt (hunk 1 above)
Comment (leharbhatt) [reply answers.txt:6]: Added the new point, committed the changes and created a new pull request.

**Pull request merged**